addons/auth_passkey/controllers/main.py

```diff
@@ -0,0 +1,34 @@
+import json
+from base64 import urlsafe_b64encode
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.web.controllers.home import Home, CREDENTIAL_PARAMS
+
+from ..lib.duo_labs.webauthn import options_to_json
+
+
+CREDENTIAL_PARAMS.append('webauthn_response')
+
+
+class WebauthnController(http.Controller):
+    @http.route(['/auth/passkey/start-registration'], type='json', auth='user')
+    def json_start_registration(self):
+        registration_options = request.env['auth.passkey.key']._create_registration_options()
+        request.session.webauthn_challenge = registration_options.challenge
+        return json.loads(options_to_json(registration_options))
+
+    @http.route(['/auth/passkey/verify-registration'], type='json', auth='user')
+    def json_verify_registration(self, registration):
+        verification = request.env['auth.passkey.key']._verify_registration_options(
+            registration,
+            request.session.pop('webauthn_challenge'),
+        )
+        return {'credentialId': urlsafe_b64encode(verification.credential_id),
+                'credentialPublicKey': urlsafe_b64encode(verification.credential_public_key)}
+
+    @http.route(['/auth/passkey/start-auth'], type='json', auth='public')
+    def json_start_authentication(self):
+        auth_options = request.env['auth.passkey.key']._start_auth()
+        request.session.webauthn_challenge = auth_options.challenge
+        return json.loads(options_to_json(auth_options))
```

Comment:
Thanks to all the work which was done, including moving the logic regarding the calls to the webauthn library inside the model methods, I think we should get rid of the 2 routes `/start-registration'` and `/verify-registration`, to keep only the route `/auth/passkey/start-auth`, which is the only one really mandatory as it must be called as `auth='public'`, without user signed in.

For the two others, you need a user to use them, even if we extend the feature to portal (you need an account before being able to create a pass key), so they can call the XMLRPC API. Only for the auth you cannot call the XMLRPC API, as you need a signed in user to be able to call the XMLRPC API.

The only case where we would need to keep the `/start-registration` would be if one day we allow to sign up with a passkey directly, but this is out of scope for the moment.
And even then, you could get rid of the `/verify-registration`, as you could call it internally from the POST request sent on signup.

The content of `/verify-registration`, `_verify_registration_options` can definitely be called inside the `make_key` of `auth.passkey.key.name`. This offers 3 benefits:
- We get rid of one route: Less route, less code, less APIs possibility for attackers, the better
- We get rid of one RPC call: Less rpc calls, the better.
  - Before, it was one RPC call to the route `/verify-registration` followed by one call to `make_key`, now it will be only one call to `make_key`, to which you pass the `registration` variable content, `const registration = await startRegistration`, instead of the `verification` variable content `const verification = await this.rpc("/auth/passkey/verify-registration"`. And it's `make_key` which become in charge to do the internal call to `_verify_registration_options`
-  The credential identifier and the public key will be validated, by  the webauthn library call `verify_registration_response`, while before we were just inserting the credential identifier and the public key without any validation directly in the database, giving the possibility to the user to shoot himself in the foot or the attacker an attack vector.

It's less evident for replacing `/start-registration`. Most-likely what should be done is to add a new field to the wizard `auth.passkey.key.name` which contains the content of the call to `_create_registration_options`, which you add in the form view but hidden, and then the javascript code gets the options it must sent to `startRegistration` from that hidden field/input instead of doing a Javascript RPC call to the former route.
This offer 2 benefits:
- We get rid of one route: Less route, less code, less APIs possibility for attackers, the better
- We get rid of one RPC call, the less RPC call, the better.
But one downside:
- One field more on the wizard.

That being said, we could also consider to pass the result of `_create_registration_options` to the dict returned by `action_create_passkey`, for instance in the context, if we want to avoid that additional field.

In the end, if we get rid of both, we reduce from 3 to 1 RPC call during the registration wizard: you only have the RPC call to create the wizard, and you get rid of the 2 calls to the routes `/start-registration`, `/verify-registration`.

I am not formely against keeping `/start-registration`, but `/verify-registration` can definitely be removed.